> Could you please look into the regression on osx?

From what I can tell, this doesn't seem to be a regression (at least, not caused by my changes). Using a clean clone of Microsoft/vcpkg, with existing tinyxml2, I get the same error:

```
$ ./vcpkg install ignition-msgs5
Computing installation plan...
The following packages will be built and installed:
  * eigen3[core]:x64-osx
  * ignition-cmake2[core]:x64-osx
  * ignition-math6[core]:x64-osx
  * ignition-modularscripts[core]:x64-osx
    ignition-msgs5[core]:x64-osx
  * protobuf[core]:x64-osx
  * tinyxml2[core]:x64-osx

...

Starting package 7/7: ignition-msgs5:x64-osx
Building package ignition-msgs5[core]:x64-osx...
-- Downloading https://github.com/ignitionrobotics/ign-msgs/archive/ignition-msgs5_5.1.0.tar.gz...
-- Extracting source /Users/orudge/vcpkg/vcpkg/downloads/ignitionrobotics-ign-msgs-ignition-msgs5_5.1.0.tar.gz
-- Applying patch 01-protobuf.patch
-- Using source at /Users/orudge/vcpkg/vcpkg/buildtrees/ignition-msgs5/src/sgs5_5.1.0-8f48d4cbaf
-- Configuring x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: /Users/orudge/vcpkg/vcpkg/downloads/tools/cmake-3.17.2-osx/cmake-3.17.2-Darwin-x86_64/CMake.app/Contents/bin/cmake /Users/orudge/vcpkg/vcpkg/buildtrees/ignition-msgs5/src/sgs5_5.1.0-8f48d4cbaf -DBUILD_TESTING=OFF -DCMAKE_MAKE_PROGRAM=/Users/orudge/vcpkg/vcpkg/downloads/tools/ninja-1.10.0-osx/ninja -DCMAKE_SYSTEM_NAME=Darwin -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/orudge/vcpkg/vcpkg/scripts/toolchains/osx.cmake -DVCPKG_TARGET_TRIPLET=x64-osx -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/Users/orudge/vcpkg/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/Users/orudge/vcpkg/vcpkg/packages/ignition-msgs5_x64-osx/debug
    Working Directory: /Users/orudge/vcpkg/vcpkg/buildtrees/ignition-msgs5/x64-osx-dbg
    Error code: 1
    See logs for more information:
      /Users/orudge/vcpkg/vcpkg/buildtrees/ignition-msgs5/config-x64-osx-dbg-out.log
      /Users/orudge/vcpkg/vcpkg/buildtrees/ignition-msgs5/config-x64-osx-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:306 (vcpkg_execute_required_process)
  installed/x64-osx/share/ignitionmodularscripts/ignition_modular_library.cmake:3 (vcpkg_configure_cmake)
  installed/x64-osx/share/ignitionmodularscripts/ignition_modular_library.cmake:131 (ignition_modular_build_library)
  ports/ignition-msgs5/portfile.cmake:13 (ignition_modular_library)
  scripts/ports.cmake:90 (include)


Error: Building package ignition-msgs5:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: ignition-msgs5:x64-osx
  Vcpkg version: 2020.02.04-unknownhash
```

It looks like the ignition-msgs5 port is looking for a non-existent vcpkg/packages/tinyxml2_x64-osx/debug/include folder - no other packages have a debug/include folder though, and I'm not quite sure where this is coming from for tinyxml2. Unfortunately I don't think this is something I have the time (or necessarily expertise) to debug further.

--- a/ports/tinyxml2/CONTROL
+++ b/ports/tinyxml2/CONTROL
@@ -2,4 +2,3 @@ Source: tinyxml2
 Version: 8.0.0
 Homepage: https://github.com/leethomason/tinyxml2
 Description: A simple, small, efficient, C++ XML parser
-

--- a/ports/tinyxml2/CONTROL
+++ b/ports/tinyxml2/CONTROL
@@ -1,5 +1,5 @@
 Source: tinyxml2
-Version: 7.1.0
+Version: 8.0.0
 Homepage: https://github.com/leethomason/tinyxml2
 Description: A simple, small, efficient, C++ XML parser
 

--- a/ports/tinyxml2/fix-symbol-exports.patch
+++ b/ports/tinyxml2/fix-symbol-exports.patch
@@ -1,0 +1,36 @@
+From f8b048656b9937252ce5fb4ebea9e5df5b203bdb Mon Sep 17 00:00:00 2001
+From: Owen Rudge <owen@owenrudge.net>
+Date: Wed, 27 May 2020 09:40:59 +0100
+Subject: [PATCH] cmake: Don't export symbols if building static library
+
+---
+ CMakeLists.txt | 12 +++++++++---
+ 1 file changed, 9 insertions(+), 3 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 9f9ab07..67b4f33 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -50,10 +50,16 @@ set(CMAKE_DEBUG_POSTFIX "d")
+ 
+ add_library(tinyxml2 tinyxml2.cpp tinyxml2.h)
+ 
+-set_target_properties(tinyxml2 PROPERTIES
++if(BUILD_SHARED_LIBS)
++    set_target_properties(tinyxml2 PROPERTIES
+         COMPILE_DEFINITIONS "TINYXML2_EXPORT"
+-	VERSION "${GENERIC_LIB_VERSION}"
+-	SOVERSION "${GENERIC_LIB_SOVERSION}")
++        VERSION "${GENERIC_LIB_VERSION}"
++        SOVERSION "${GENERIC_LIB_SOVERSION}")
++else()
++    set_target_properties(tinyxml2 PROPERTIES
++        VERSION "${GENERIC_LIB_VERSION}"
++        SOVERSION "${GENERIC_LIB_SOVERSION}")
++endif()
+ 
+ target_compile_definitions(tinyxml2 PUBLIC $<$<CONFIG:Debug>:TINYXML2_DEBUG>)
+ 
+-- 
+2.25.1.windows.1
+

--- a/ports/tinyxml2/portfile.cmake
+++ b/ports/tinyxml2/portfile.cmake
@@ -3,9 +3,11 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO leethomason/tinyxml2
-    REF 9c740e8d2341bd46dbe8e87053cdb4d931971967 # 7.1.0
-    SHA512 a0e9634875f4c5f426f41510040b9f078af24adf176d2daf3cb3343d629b8068f3a1841df80a06d977bd19e3acaaa3736719a900754c1fe675631f3337820130
+    REF 8.0.0
+    SHA512 bcbb065c2af34ea681ec556377fd22e720b6f5d4caa73f432b1e34e08603a96f2233763f0ec5ae86b9ee71ddbe3062f58d3794cd3a162ce6903435530de0bba6
     HEAD_REF master
+    PATCHES
+        fix-symbol-exports.patch
 )
 
 vcpkg_configure_cmake(

--- a/ports/tinyxml2/portfile.cmake
+++ b/ports/tinyxml2/portfile.cmake
@@ -1,5 +1,3 @@
-include(vcpkg_common_functions)
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO leethomason/tinyxml2
@@ -24,8 +22,4 @@ vcpkg_copy_pdbs()
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
 
-file(COPY
-  ${SOURCE_PATH}/readme.md
-  DESTINATION ${CURRENT_PACKAGES_DIR}/share/tinyxml2
-)
-file(RENAME ${CURRENT_PACKAGES_DIR}/share/tinyxml2/readme.md ${CURRENT_PACKAGES_DIR}/share/tinyxml2/copyright)
+file(INSTALL ${SOURCE_PATH}/readme.md DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

